assets/js/base/context/hooks/collections/use-collection.ts

```diff
@@ -77,9 +79,23 @@ export const useCollection = (
 	const throwError = useThrowError();
 	const results = useSelect(
 		( select ) => {
+			if (
+				isEditor &&
+				typeof currentResults === 'object' &&
+				currentResults.hasOwnProperty( 'current' ) &&
+				currentResults.current.hasOwnProperty( 'results' ) &&
```

Comment:
We can use `isObject` from `@woocommerce/types` here to make this call shorter, something like below. It also fixes the current TS error:

<img width="914" alt="image" src="https://user-images.githubusercontent.com/5423135/210086550-19710fab-b625-4692-a8f4-4376db622c3b.png">


```suggestion
				isObject( currentResults?.current?.results ) &&
```

Reply:
Good call, updated!